tests/lib/rules/no-classic-classes.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * @fileoverview Disallow "classic" classes in favor of native JS classes
+ * @author Alex LaFroscia
+ */
+'use strict';
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require('../../../lib/rules/no-classic-classes');
+const RuleTester = require('eslint').RuleTester;
+
+const { ERROR_MESSAGE_NO_CLASSIC_CLASSES: ERROR_MESSAGE } = rule;
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parserOptions: { ecmaVersion: 6, sourceType: 'module' },
+});
+
+ruleTester.run('no-classic-classes', rule, {
+  valid: [
+    `
+      import Component from '@ember/component';
+
+      export default class MyComponent extends Component {}
+    `,
+    `
+      import Controller from '@ember/controller';
+
+      export default class MyController extends Controller {}
+    `,
+    `
+      import Route from '@ember/routing/route';
+
+      export default class MyRoute extends Route {}
+    `,
+    `
+      import Component from '@ember/component';
+
+      export default class MyRoute extends Route.extend(SomeMixin) {}
+    `,
```

Comment:
Can you add valid examples to ensure that other types of objects are allowed to be created still?

Reply:
Do you mean some object that's totally unrelated to Ember, but that has an `.extend` method?

Reply:
Assuming that's what you meant, I wrote a test case for that. I'm glad you mentioned it -- the rule complained about any call to `.extend`!